man/ctags-optlib.7.rst.in

```diff
@@ -285,14 +294,110 @@ representation. ``--list-regex-flags`` lists all the flags.
 	manner.
 
 ``placeholder``
-	TBW
+	Don't emit a tag captured with a regex pattern.  The replacement
+	can be an empty string.  See the following description of
+	``scope=...`` flag about how this is useful.
+
+``scope=ref|push|pop|clear|set``
+
+	Specify what to do with the internal scope stack.
+
+	A parser programmed with ``--regex-<LANG>`` has a stack (scope
+	stack) internally. You can use it for tracking scope
+	information. The ``scope=...`` flag is for manipulating and
+	utilizing the scope stack.
+
+	If ``{scope=push}`` is specified, a tag captured with
+	``--regex-<LANG>`` is pushed to the stack. ``{scope=push}``
+	implies ``{scope=ref}``.
+
+	You can fill the scope field of captured tag with
+	``{scope=ref}``. If ``{scope=ref}`` flag is given,
+	@CTAGS_NAME_EXECUTABLE@ attaches the tag at the top to the tag
+	captured with ``--regex-<LANG>`` as the value for the ``scope:``
+	field.
+
+	@CTAGS_NAME_EXECUTABLE@ pops the tag at the top of the stack when
+	``--regex-<LANG>`` with ``{scope=pop}`` is matched to the input
+	line.
+
+	Specifying ``{scope=clear}`` removes all the tags in the scope.
+	Specifying ``{scope=set}`` removes all the tags in the scope, and
+	then pushes the captured tag as ``{scope=push}`` does.
+
+	In some case, you may want to use ``--regex-<LANG>`` only for side
+	effect: using only for manipulating the stack but not for
+	capturing a tag. In such case, make *replacement* component of
+	``--regex-<LANG>`` option empty with specifying ``{placeholder}``
+	as a regex flag. For an example, you can put a non-named tag to
+	the stack by giving a regex flag ``{scope=push}{placeholder}``.
+
+	You may wonder what happens if a regex patter having
```

Comment:
`s/patter having/pattern with a/

Reply:
``{scope=ref}`` follows the sentence, so I think "pattern with " is better than "pattern with a".
Am I wrong.

----


You may wonder what happens if a regex pattern with ``{scope=ref}`` flag ...

or


You may wonder what happens if a regex pattern with a ``{scope=ref}`` flag ...

Reply:
It sounds more natural with an article in that position because `{scope=ref} flag` is a countable noun. I think technically "the" is the correct article because `{scope=ref} flag` is definite, but "a" sounds just as good to me here.

----

A perhaps more clear way to structure the ideas might be like this:

> If the regex pattern contains a ``{scope=ref}`` flag and the stack is empty, the ``{scope=ref}`` flag is ignored and nothing is attached to the ``scope:`` field. 

> If the top of the stack contains an unnamed tag, @CTAGS_NAME_EXECUTABLE@ searches deeper into the stack to find the top-most named tag. If it reaches the bottom of the stack without finding a named tag, the ``{scope=ref}`` flag is ignored and nothing is attached to the
``scope:`` field.

I don't know if that's technically correct, but it's how I understood it as a user.

Reply:
Yes, that's what I wanted to write. I should split the description into two as you did.